### PR TITLE
[`flake8-logging`] Implement `LOG007`: `ExceptionWithoutExcInfo`

### PR DESCRIPTION
**Pull request by qdegraaf**
## Summary

This PR implements a new rule for `flake8-logging` plugin that checks for uses of `logging.exception()` with `exc_info` set to `False` or a falsy value. It suggests using `logging.error` in these cases instead. 

I am unsure about the name. Open to suggestions there, went with the most explicit name I could think of in the meantime.

Refer https://github.com/astral-sh/ruff/issues/7248

## Test Plan

Added a new fixture cases and ran `cargo test`

### CODE REVIEW

File: crates/ruff/src/rules/flake8_logging/rules/exc_info_false_in_exception.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        format!("Use of `logging.exception` with falsy `exc_info`")
+    }
+}
+
+/// LOG007
+pub(crate) fn exc_info_false_in_exception(checker: &mut Checker, call: &ExprCall) {
+    if checker
+        .semantic()
+        .resolve_call_path(call.func.as_ref())
+        .is_some_and(|call_path| matches!(call_path.as_slice(), ["logging", "exception"]))
```
Comment (qdegraaf) [inline crates/ruff/src/rules/flake8_logging/rules/exc_info_false_in_exception.rs:43]: Would it be worth it to check the name of the call first before calling `resolve_call_path()`for performance reasons? Or would that just add unnecessary clutter?

File: crates/ruff/src/rules/flake8_logging/rules/exc_info_false_in_exception.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for uses of `logging.exception()` with `exc_info` set to `False`
+///
+/// ## Why is this bad?
+/// The `exception()` method captures the exception automatically. Disabling this by setting
+/// `exc_info=False` is the same as using `error()`, which is clearer and doesn’t need the
+/// `exc_info`argument. This rule detects `exception()` calls with an exc_info argument that is
```
Comment (konstin) [inline crates/ruff/src/rules/flake8_logging/rules/exc_info_false_in_exception.rs:16]:
```suggestion
/// `exc_info=False` is the same as using `error()`, which is clearer and doesn't need the
/// `exc_info` argument. This rule detects `exception()` calls with an exc_info argument that is
```

File: crates/ruff/resources/test/fixtures/flake8_logging/LOG007.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import logging
```
Comment (konstin) [inline crates/ruff/resources/test/fixtures/flake8_logging/LOG007.py:1]: Can you add a test case with `logger.exception`? (https://github.com/adamchainz/flake8-logging/blob/8cd23181647d4c68391af1db71588e8b3dc2bcfe/tests/test_flake8_logging.py#L783-L793)

File: crates/ruff/src/rules/flake8_logging/rules/exc_info_false_in_exception.rs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+impl Violation for ExcInfoFalseInException {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Use of `logging.exception` with falsy `exc_info`")
+    }
+}
+
+/// LOG007
+pub(crate) fn exc_info_false_in_exception(checker: &mut Checker, call: &ExprCall) {
+    if checker
```
Comment (konstin) [inline crates/ruff/src/rules/flake8_logging/rules/exc_info_false_in_exception.rs:40]: This needs to catch all log messages.

More common than
```python
logging.exception("...")
```
is doing
```python
logger = logging.getLogger(__name__)

logger.exception("...")
```
ruff has utils to detect logging calls, see `is_logger_candidate` and e.g. the G001 implementation.

Comment (qdegraaf) [reply crates/ruff/src/rules/flake8_logging/rules/exc_info_false_in_exception.rs]: Rewrote to utilise `is_logger_candidate`. It flags `logger.exception` and `logging.exception` now but does not flag `from logging import exception; exception()` Is that a limitation of `is_logger_candidate`? And if so, do we want to create another branch of logic to catch those cases?

**Review by charliermarsh — APPROVED** [review]

File: crates/ruff/src/rules/flake8_logging/rules/exc_info_false_in_exception.rs (hunk 4 above)
Comment (konstin) [reply crates/ruff/src/rules/flake8_logging/rules/exc_info_false_in_exception.rs]: Moved this to https://github.com/astral-sh/ruff/issues/7502

**Pull request merged**